texar/data/data/multi_aligned_data.py

```diff
@@ -220,35 +217,48 @@ def _get_filter(delimiter, max_seq_length):
                                                hparams_i["max_seq_length"]))
                 else:
                     filters.append(None)
+
+                if _is_text_data(dtype):
+                    self._names.append(
+                        {"text": connect_name(hparams_i["data_name"], "text"),
+                         "text_ids": connect_name(hparams_i["data_name"],
+                                                  "text_ids"),
+                         "length": connect_name(hparams_i["data_name"],
+                                                "length")})
+
+                    text_hparams = MonoTextData.default_hparams()
+                    text_hparams.update({"dataset": hparams_i.todict()})
+
+                    # MonoTextData does not have following keys
+                    for key in ["data_type", "vocab_share_with",
```

Comment:
Instead of hard-coding keys, why not just iterate over all keys in `default_hparams`, and only updating those that are valid?

Reply:
Done! [462312cab452a7f1c3eacce634d65f1a50c2fbbc]